### PR TITLE
Update user list membership table

### PR DESCRIPTION
**Pull request by sophiemachin**
Rename `anon_project_specific_user_id` to `aps_or_real_user_id`

Remove foreign key

Give it a UUID default

### CODE REVIEW

**Comment by this-andy** [comment]
https://this.tpondemand.com/entity/4873

File: thiscovery_admin/projects/migrations/0083_auto_20230412_1154.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    operations = [
+        migrations.RemoveField(
+            model_name='userlistmembership',
+            name='anon_project_specific_user',
+        ),
+        migrations.AddField(
+            model_name='userlistmembership',
+            name='aps_or_real_user_id',
+            field=models.UUIDField(default=uuid.uuid4),
```
Comment (this-andy) [inline thiscovery_admin/projects/migrations/0083_auto_20230412_1154.py:21]: do we need a default here?  The creating process should always provide it and if it doesn't then we should throw an error, not default it.  So I'd suggest omitting default, but leaving as non null

File: thiscovery_admin/projects/models.py [hunk 2]
```diff
@@ -473,7 +473,4 @@ def number_of_users(self):
 
 class UserListMembership(TimeStampedModel):
     user_list = models.ForeignKey(UserList, on_delete=models.CASCADE)
-    anon_project_specific_user = models.ForeignKey(
-        UserProject, on_delete=models.CASCADE,
-        to_field="anon_project_specific_user_id",
-    )
+    aps_or_real_user_id = models.UUIDField(default=uuid.uuid4)
```
Comment (this-andy) [inline thiscovery_admin/projects/models.py:476]: As above

File: thiscovery_admin/projects/migrations/0083_auto_20230412_1154.py (hunk 1 above)
Comment (sophiemachin) [reply thiscovery_admin/projects/migrations/0083_auto_20230412_1154.py:21]: "You are trying to add a non-nullable field 'aps_or_real_user_id' to userlistmembership without a default; we can't do that (the database needs something to populate existing rows)."

**Pull request closed without merging**